src/main/java/org/jitsi/jigasi/transcription/OracleTranscriptionService.java

```diff
@@ -0,0 +1,350 @@
+{
+
+    /**
+     * Uses the PCMAudioSilenceMediaDevice which transcodes to 16khz PCM audio
+     *
+     * @param listener
+     * @return AudioMixerMediaDevice
+     */
+    @Override
+    public AudioMixerMediaDevice getMediaDevice(ReceiveStreamBufferListener listener)
+    {
+        if (this.mediaDevice == null)
+        {
+            this.mediaDevice = new TranscribingAudioMixerMediaDevice(new PCMAudioSilenceMediaDevice(), listener);
+        }
+
+        return this.mediaDevice;
+    }
+
+    /**
+     * The logger for this class
+     */
+    private final static Logger logger
+            = Logger.getLogger(OracleTranscriptionService.class);
+
+    /**
+     * The config key of the websocket to the speech-to-text service.
+     */
+    public final static String WEBSOCKET_URL
+            = "org.jitsi.jigasi.transcription.oci.websocketUrl";
+
+    public final static String OCI_AUTH_CONFIGURATION_FILE
+            = "org.jitsi.jigasi.transcription.oci.configFile";
+
+    public final static String defaultOCIAuthConfFile = "~/.oci/config";
+
+    private static final String configFilePath = JigasiBundleActivator.getConfigurationService()
+            .getString(OCI_AUTH_CONFIGURATION_FILE, defaultOCIAuthConfFile);
+
+    public final static String COMPARTMENT_ID
+            = "org.jitsi.jigasi.transcription.oci.compartmentId";
+
+    private final String compartmentId;
+
+    public final static String DEFAULT_WEBSOCKET_URL = "ws://localhost:8000/ws";
+
+    private static BasicAuthenticationDetailsProvider authProvider;
+
+    static
+    {
+
+        try
+        {
+            authProvider = new ConfigFileAuthenticationDetailsProvider(ConfigFileReader.parse(configFilePath));
+        }
+        catch (IOException e)
+        {
+            logger.warn("Error while reading OCI configuration file, trying to use the instance's principal", e);
+        }
+
+        if (authProvider == null)
+        {
+            try
+            {
+                authProvider = new InstancePrincipalsAuthenticationDetailsProvider.
+                        InstancePrincipalsAuthenticationDetailsProviderBuilder().
+                        build();
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while creating OCI instance principal provider", e);
+            }
+        }
+    }
+
+
+    /**
+     * The config value of the websocket to the speech-to-text service.
+     */
+    private final String websocketUrlConfig;
+
+
+    /**
+     * Create a TranscriptionService which will send audio to the OCI service
+     * platform to get a transcription
+     */
+    public OracleTranscriptionService()
+    {
+        websocketUrlConfig = JigasiBundleActivator.getConfigurationService()
+                .getString(WEBSOCKET_URL, DEFAULT_WEBSOCKET_URL);
+        compartmentId = JigasiBundleActivator.getConfigurationService()
+                .getString(COMPARTMENT_ID, "");
```

Comment:
Maybe don't default to empty string, but log an error if null so it is known that something is missing and nothing will work?

Reply:
logs an error now